src/containers/WelcomePage/components/LastUsedContent.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { FC, useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { injectT } from '@ndla/i18n';
+
+import {
+  ArticleType,
+  ContentResultType,
+  ResourceType,
+  TranslateType,
+} from '../../../interfaces';
+import { RESOURCE_TYPE_LEARNING_PATH } from '../../../constants';
+import {
+  getContentTypeFromResourceTypes,
+  resourceToLinkProps,
+} from '../../../util/resourceHelpers';
+import { isLearningpath } from '../../../util/routeHelpers';
+import formatDate from '../../../util/formatDate';
+import { fetchDraft } from '../../../modules/draft/draftApi';
+import { classes } from '../WelcomePage';
+
+interface Props {
+  articleId: number;
+  content: ContentResultType;
+  locale: string;
+  t: TranslateType;
+}
+
+const LastUsedContent: FC<Props> = ({ articleId, content, locale, t }) => {
+  const { contexts } = content;
+
+  const [article, setArticle] = useState<ArticleType | undefined>(undefined);
+
+  const fetchArticle = async (articleId: number, locale: string) => {
+    const article = await fetchDraft(articleId, locale);
+    setArticle(article);
+  };
+
+  useEffect(() => {
+    if (articleId) {
+      fetchArticle(articleId, locale);
+    }
+  }, []);
+
+  let resourceType: ResourceType | any = {};
```

Comment:
@N4G1 Denne er jeg usikker på, men har ikke klart å finne en bra løsning. Har du noe forslag?

Reply:
Jeg ville satt hele den kodesnutten fra `let resourceType:...` frem til  `const linkProps = resourceToLinkProps(` (ikke inkludert) i en funksjon og kalt den funksjonen inn i useEffect()

Reply:
Noe slikt tenker jeg: `  const getResourceType = () => {
    const result = content.contexts?.find(a => a.resourceTypes)?.resourceTypes;
    if(result) {
      return getContentTypeFromResourceTypes(result);
    } else {
      return isLearningpath(content.url) && getContentTypeFromResourceTypes([
        { id: RESOURCE_TYPE_LEARNING_PATH },
      ]);
    }
  };`

Reply:
og ikke kall det i useEffect() for det ikke der den blir brukt men inn i linkProps

Reply:
Helst dobbeltsjekk funksjonalitet og de diverse use-casene

Reply:
Føler det var letter å lese koden slik det stod før, selv om denne har mer "fancy" syntax. Hva er result og hvor kommer denne fra? Og hva står a for, hva er a i contexts?